src/net.h

```diff
@@ -392,6 +407,27 @@ class CNode
     uint32_t nXthinBloomfilterSize; // The maximum xthin bloom filter size (in bytes) that our peer will accept.
     // BUIP010 Xtreme Thinblocks: end section
 
+    // BUIPXXX Graphene blocks: begin section
+    CCriticalSection cs_mempoolsize;
+    CBlock grapheneBlock;
```

Comment:
what is this critical section locking...i would assume everything underneath it but seems like a lot of things there that maybe are not related?  not sure...  if not everything underneath, then separate items out... at a minimum have the criticalsection first, then items it locks, then an empty line.

Reply:
It only locks only `nGrapheneMemPoolTx`, but I realize now that I didn't follow naming conventions for locks. I renamed it to `cs_ngraphenemempooltx`, grouped it with `nGrapheneMemPoolTx`, and added a comment.

I'm also not certain that a lock is necessary here. I was following the lead of `LoadFilter`, which locks `cs_filter`, performs a delete, and then reallocates memory. However, in my case I'm just updating an integer, so perhaps the lock is overkill. My ambivalence is really a result of not knowing the broader codebase better. Is it the case that multiple block requests from the same node could be processed simultaneously? Thinking about it now, I figure not and the cs_filter lock was just a precaution. 

In general I think I need to do a lock audit. Perhaps at some point I can get with someone who knows the code a little better and talk through where there is concurrency and what objects are affected.